apps/trading/components/rewards-container/streaks/reward-hoarder-bonus.tsx

```diff
@@ -0,0 +1,184 @@
+import { VegaIcon, VegaIconNames } from '@vegaprotocol/ui-toolkit';
+import { useT } from '../../../lib/use-t';
+import classNames from 'classnames';
+import type { PartyVestingStats } from '@vegaprotocol/types';
+import BigNumber from 'bignumber.js';
+import { formatNumber } from '@vegaprotocol/utils';
+
+export const RewardHoarderBonus = ({
+  tiers,
+  vestingDetails,
+}: {
+  tiers:
+    | never[]
+    | {
+        minimum_quantum_balance: string;
+        reward_multiplier: string;
+      }[];
+  vestingDetails?: PartyVestingStats | null;
+}) => {
+  const t = useT();
+  if (!vestingDetails) return null;
+
+  const getUserTier = () => {
+    let userTier = 0,
+      i = 0;
+    do {
+      userTier = i;
+      i++;
+    } while (
+      i < tiers.length &&
+      new BigNumber(tiers[userTier].minimum_quantum_balance).isLessThan(
+        vestingDetails.quantumBalance
+      )
+    );
+    if (
+      i === tiers.length &&
+      new BigNumber(
+        tiers[userTier].minimum_quantum_balance
+      ).isLessThanOrEqualTo(vestingDetails.quantumBalance)
+    ) {
+      userTier = i;
+    }
+    return userTier;
+  };
+  if (!tiers || tiers.length === 0) return null;
+  const userTierIndex = getUserTier();
+
+  // There is only value to compare to the tiers that covers all the user' rewards across all assets
+  const qUSD = 'qUSD';
+
+  const safeProgress = (i: number) => {
+    if (i < userTierIndex) return 100;
+    if (i > userTierIndex) return 0;
+    const progress = vestingDetails.quantumBalance;
+    const total = tiers[i].minimum_quantum_balance;
+    if (new BigNumber(progress).isGreaterThan(total)) return 100;
+    return new BigNumber(progress)
+      .multipliedBy(100)
+      .dividedBy(total)
```

Comment:
The case for infinity and beyond.
**question:** Could this be 0 (zero)?

Reply:
good catch! i will add an extra check now